watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskAdapter.java

```diff
@@ -5,64 +5,119 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ArrayAdapter;
+import android.widget.BaseExpandableListAdapter;
 import android.widget.TextView;
 
 import com.blueprint.watershed.R;
 
-import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
 
 /**
  * A Custom Adapter for the Task Object
  * Code heavily reference from http://www.ezzylearning.com/tutorial/customizing-android-listview-items-with-custom-arrayadapter
  *
  */
-public class TaskAdapter extends ArrayAdapter<Task> {
-    Context context;
-    int layoutResourceId;
-    ArrayList<Task> data = null;
-
-    public TaskAdapter(Context context, int layoutResourceId, ArrayList<Task> data){
-        super(context, layoutResourceId, data);
-        this.layoutResourceId = layoutResourceId;
-        this.context = context;
-        this.data = data;
+public class TaskAdapter extends BaseExpandableListAdapter {
+
+    private Context mContext;
+    private List<String> mHeaders;
+    private HashMap<String, List<Task>> mData;
+
+    public TaskAdapter(Context context, List<String> headers, HashMap<String, List<Task>> data){
+        super();
+        mContext = context;
+        mHeaders = headers;
+        mData = data;
+    }
+
+    @Override
+    public int getChildrenCount(int groupPosition) {
+        List<Task> tasks = mData.get(mHeaders.get(groupPosition));
+        return tasks == null ? 0 : tasks.size();
     }
 
     @Override
-    public View getView(int position,  View convertview, ViewGroup parent){
-        View row = convertview;
-        TaskHolder holder = null;
+    public boolean isChildSelectable(int groupPosition, int childPosition) { return true; }
+
+    @Override
+    public long getChildId(int groupPosition, int childPosition) { return childPosition; }
```

Comment:
I know, but if we have more than 1 group, then we can't just return childPosition, because we have to account for the groups too.